### PR TITLE
feat: change error casting in go113

### PR DESCRIPTION
**Pull request by AlphaWong**
# objective
seems go 1.13 introduces a better way for error casting should we change the statement like my commit ?

# reason
seems the minimum go version is already >= 1.13

https://github.com/cyrilgdn/terraform-provider-rabbitmq/blob/master/.travis.yml#L7 

# reference
https://github.com/golang/go/wiki/ErrorValueFAQ

### CODE REVIEW

File: rabbitmq/util.go [hunk 1]
```diff
@@ -9,14 +10,13 @@ import (
 )
 
 func checkDeleted(d *schema.ResourceData, err error) error {
-	switch e := err.(type) {
-	case rabbithole.ErrorResponse:
-		if e.StatusCode == 404 {
+	var errorResponse *rabbithole.ErrorResponse
```
Comment (cyrilgdn) [inline rabbitmq/util.go:13]:
```suggestion
	var errorResponse rabbithole.ErrorResponse
```

**Review by cyrilgdn — CHANGES_REQUESTED** [review]
Hi @AlphaWong ,

Thanks for opening this PR. Could you push again, Travis was disabled and didn't run. I enabled it back.

Tests currently don't pass without the suggestion I made (rabbithole does not return a pointer).

We can use this new syntax but, as I'm not a Go expert, it's still quite blurry for me with this pointer to a type :thinking:

**Comment by AlphaWong** [comment]
> Hi @AlphaWong ,
> 
> Thanks for opening this PR. Could you push again, Travis was disabled and didn't run. I enabled it back.
> 
> Tests currently don't pass without the suggestion I made (rabbithole does not return a pointer).
> 
> We can use this new syntax but, as I'm not a Go expert, it's still quite blurry for me with this pointer to a type 🤔

thx a lot

**Review by cyrilgdn — APPROVED** [review]
Thanks :+1: 

I merged master into your branch to fix the other error in the tests (was still executing some Hashicorp tests for the previous repository)

**Pull request merged**